nativelib/src/main/scala/java/lang/Object.scala

```diff
@@ -18,8 +19,18 @@ class _Object {
   @inline def __toString(): String =
     getClass.getName + "@" + Integer.toHexString(hashCode)
 
-  @inline def __getClass(): _Class[_] =
-    new _Class(runtime.getRawType(this))
+  @inline def __getClass(): _Class[_] = {
+    val rtti   = getRawType(this)
+    val clsPtr = elemRawPtr(rtti, 16)
+
+    if (loadRawPtr(clsPtr) == null) {
+      val newClass = new _Class[Any](rtti)
+      storeObject(clsPtr, newClass)
+      ClassInstancesRegistry.add(newClass)
+    } else {
+      loadObject(clsPtr).asInstanceOf[_Class[_]]
+    }
```

Comment:
I suggest moving all these low-level manipulations to a new method `runtime.toClass(rawType: RawPtr): Class[_]`, so that this method becomes simply
```scala
runtime.toClass(runtime.getRawType(this))
```
that way we keep the runtime-related concepts inside `runtime.*`.